lightning/src/routing/router.rs

```diff
@@ -5465,12 +5465,20 @@ mod benches {
 			}
 		}
 
+		// Because we've changed channel scores, its possible we'll take different routes to the
+		// selected destinations, possibly causing us to fail because, eg, the newly-selected path
+		// requires a too-high CLTV delta.
+		route_endpoints.retain(|(first_hop, params, amt)| {
+			get_route(&payer, params, &graph.read_only(), Some(&[first_hop]), *amt, 42, &DummyLogger{}, &scorer, &random_seed_bytes).is_ok()
+		});
+		route_endpoints.truncate(100);
+		assert_eq!(route_endpoints.len(), 100);
+
 		// ...then benchmark finding paths between the nodes we learned.
-		let mut idx = 0;
 		bench.iter(|| {
-			let (first_hop, params, amt) = &route_endpoints[idx % route_endpoints.len()];
-			assert!(get_route(&payer, params, &graph.read_only(), Some(&[first_hop]), *amt, 42, &DummyLogger{}, &scorer, &random_seed_bytes).is_ok());
-			idx += 1;
+			for (first_hop, params, amt) in route_endpoints.iter() {
+				assert!(get_route(&payer, params, &graph.read_only(), Some(&[first_hop]), *amt, 42, &DummyLogger{}, &scorer, &random_seed_bytes).is_ok());
+			}
```

Comment:
I believe this will significantly increase the benchmark runtime. I tried doing something similar when refactoring the benchmarks without much luck.

I have the benchmarks running on my machine now for over an hour and have only seen one result so far:

```
running 6 tests
test routing::router::benches::generate_mpp_routes_with_default_scorer       ... bench: 10,138,522,397 ns/iter (+/- 541,770,074)
```

Reply:
Grrr, I'd somehow thought the bencher was smarter than that :(.